Add endpoints related to images

ryancarlson opened the pull request:
- list images for a package
 - retrieve a specific image
 - upload a new image

ryancarlson commented:
@LaelWatkins can you review this?  I pushed it to stage and I don't see any issues w/ the existing functionality.  I've tested these specific endpoints locally too.

LaelWatkins commented:
looking at it now

ryancarlson commented:
One thing i noticed is that the filename didn't get saved.

LaelWatkins commented on src/main/java/org/cru/godtools/domain/images/ImageService.java:
```diff
@@ -20,18 +18,31 @@ public ImageService(Connection sqlConnection)
+				.addParameter("id", id)
+				.executeAndFetchFirst(Image.class);
+
+		if(image != null && image.getImageContent() != null)
```
Just a nit pick: what do you think about having a method that does this ``if``? It the same logic on line 43.

ryancarlson replied:
Good call.

LaelWatkins commented on src/main/java/org/cru/godtools/api/v2/ImageResource.java:
```diff
@@ -0,0 +1,213 @@
+	/**
+	 * Thanks to tutorial by MK Yong: http://www.mkyong.com/webservices/jax-rs/file-upload-example-in-resteasy/
+	 */
+	private String getFileName(MultivaluedMap<String, String> header)
```
Just curious, is this method is for future use?

LaelWatkins commented:
The filename is written to the ``Image`` object by the client? I'm asking because of your comment yesterday. And, it doesn't appear to be added to the object anywhere in the ``ImageResource`` class.

LaelWatkins commented:
I only have a couple of comments.  This looks great!  There is the question about the filename being set on the ``Image`` object.  Unless I'm mistaken, that doesn't appear to be an issue here in the api.  

If it does require a change, i'll take a look whenever it's ready.  Let me know.

ryancarlson commented:
@LaelWatkins I believe I've addressed your feedback here, as well as patched the cause of that NPE that cropped up.  Can you take another look when you have some time?

LaelWatkins commented:
really nice! 👍

ryancarlson commented:
Thanks @LaelWatkins
